filewatcher/filewatcher.go

```diff
@@ -255,17 +262,31 @@ type MockFileWatcher struct {
 //
 // This method is not threadsafe.
 func (fw *MockFileWatcher) Update(r io.Reader) error {
-	data, err := fw.parser(r)
+	raw, err := ioutil.ReadAll(r)
 	if err != nil {
 		return err
 	}
-	fw.data.Store(data)
+
+	data, err := fw.parser(bytes.NewReader(raw))
+	if err != nil {
+		return err
+	}
+
+	fw.data.Store(mockData{
+		raw:    raw,
+		parsed: data,
+	})
 	return nil
 }
 
+// Raw returns the raw, un-parsed data.
+func (fw *MockFileWatcher) Raw() []byte {
+	return fw.data.Load().(mockData).raw
+}
+
 // Get returns the parsed data.
 func (fw *MockFileWatcher) Get() interface{} {
-	return fw.data.Load()
+	return fw.data.Load().(mockData).parsed
```

Comment:
this will panic if `Get` is called before all `Update` calls.

Is the reason to store the struct to support single key replacement in secrets mock? that can be done by storing the whole map in mock secret struct, instead of using the filewatcher.

Reply:
> this will panic if Get is called before all Update calls.

`¯\_(ツ)_/¯ ` This is for test code and should be created using `NewMockFileWatcher`, if you don't do that and it panics that's on you IMO.

> Is the reason to store the struct to support single key replacement in secrets mock?

That's what _prompted_ it but it's to support incremental updates to the data stored in the mock file watcher in general.

Reply:
> that can be done by storing the whole map in mock secret struct, instead of using the filewatcher.

Ya, although we'd need some other changes, (like make `SecretsStore` an interface and using that, not a crazy thing to do though).

Reply:
I'll just take this out for now, we can do this in a separate PR :)